core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -755,6 +755,10 @@ public static Builder buildFrom(TableMetadata base) {
     return new Builder(base);
   }
 
+  public static Builder buildFrom(TableMetadata base, boolean noPreviousFile) {
```

Comment:
A new public api here needs a java doc, I'm not sure I follow what noPreviousFile is expected to do here

Reply:
That's a "smart thing" in the builder. If it's present, then it add it as a history item. That behavior is okay, when `buildFrom()` is only used when it is used within a "modify `TableMetadata` operation". But the Nessie catalog only uses it to modify the `TableMetadata` as it is - so it's not a change, which requires a new history-item.
Without the `noPreviousFile==true`, a couple of catalog tests fail, because those assert on the number of history items.

Reply:
@snazy I've been thinking on this and I'm wondering why can't implement a builder like  `.withPreviousFileLocation` to set that to previousFileLocation to null

Reply:
WFM, updated the PR